luajit/upipe.lua

```diff
@@ -365,6 +365,28 @@ ffi.metatype("struct uref", {
     end
 })
 
+--[[
+-- A function to iterate over all the planes of a uref.  For each plane it
+-- returns the plane string (chroma for pic, channel for sound).
+--
+-- The value for the plane will be invalid on the next iteration so make a copy
+-- if it needs to be kept.
+--]]
+local function foreach_plane(func)
+    return function(ref)
+        local plane = ffi.new("const uint8_t *[1]")
+        return function()
+            if C.ubase_check(func(ref, plane)) and plane[0] ~= nil then
+                return plane[0]
```

Comment:
Wouldn't it be more useful to return a Lua string here with `ffi.string`? Also, I believe the value is still valid on the next iteration.

Reply:
You're probably right.  I didn't want Luajit to convert back and forth since I call other upipe functions using the plane.  Probably a premature (micro)optimisation.

As for being invalid, that is probably the wrong term.  The pointer will change to point to the next plane string whereas Lua strings are immutable.  `ffi.string` is probably best.